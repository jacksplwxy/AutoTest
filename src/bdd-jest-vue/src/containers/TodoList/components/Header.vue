<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input class="header-input" type="text " data-test='header-input' v-model="inputValue"
          placeholder="请输入待完成项"
          @keyup.enter="addTodoItem"
      >
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  props: {},
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">
 .header{
   line-height  60px;
   background #ccc;
   .header-content{
     width 600px
     margin 0 auto
     .header-input{
       width 360px
       float right
       line-height 24px
       outline none
       margin-top 18px
     }
   }
 }
</style>
