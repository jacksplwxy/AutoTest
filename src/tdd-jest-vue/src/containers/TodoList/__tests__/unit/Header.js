import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'
import { findTestWrapper } from '../../../../utils/testUtils'

it('Header样式发生改变做提示', () => {
  const wrapper = shallowMount(Header)
  expect(wrapper).toMatchSnapshot()
})

it('Header包含input框', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  expect(input.exists()).toBe(true)
})

it('Header包含input框初始内容为空', () => {
  const wrapper = shallowMount(Header)
  const inputValue = wrapper.vm.$data.inputValue
  expect(inputValue).toBe('')
})

it('Header包含input框值发生变化，数据应该跟着变化', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('jacksplwxy')
  const inputValue = wrapper.vm.$data.inputValue
  expect(inputValue).toBe('jacksplwxy')
})

it('Header包含input框输入回车，无内容时无反应', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeFalsy()
})

it('Header包含input框输入回车，有内容时向外触发事件,同时清空input框', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('jacksplwxy')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
  const inputValue = wrapper.vm.$data.inputValue
  expect(inputValue).toBe('')
})
