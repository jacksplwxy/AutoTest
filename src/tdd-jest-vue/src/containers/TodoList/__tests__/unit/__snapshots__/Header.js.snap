// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header样式发生改变做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input type="text " data-test="input" placeholder="请输入待完成项" class="header-input">
  </div>
</div>
`;
